src/Expr.cc

```diff
@@ -4319,6 +4320,52 @@ void CallExpr::ExprDescribe(ODesc* d) const
 		args->Describe(d);
 	}
 
+LambdaExpr::LambdaExpr(std::unique_ptr<function_ingredients> ingredients,
+		       std::shared_ptr<id_list> outer_ids)
+	{
+	this->ingredients = std::move(ingredients);
+	this->outer_ids = std::move(outer_ids);
+	SetType(this->ingredients->id->Type()->Ref());
+	}
+
+Val* LambdaExpr::Eval(Frame* f) const
+	{
+	BroFunc* lamb = new BroFunc(
+		ingredients->id,
+		ingredients->body,
+		ingredients->inits,
+		ingredients->frame_size,
+		ingredients->priority);
+
+	lamb->AddClosure(outer_ids, f);
+
+	ingredients->id->SetVal((new Val(lamb))->Ref());
```

Comment:
`new' already refs.

Reply:
A question: Do we need to set the ID's value here, rather than just passing the new value back directly? I'm not quite sure but something here doesn't feel quite right: we keep modifying the value of the ID each time the lambda function gets executed. That means one can't really rely on the ID's value, and it's a temporary ID that's not visible to a script anyways.

Reply:
I think this comes down to the principal of least astonishment. Will people expect that the id has a value equal to the last function that it generated, or that it has none at all? Now that I think about it more, and given that it threw you a little, I figure it would probably be best just not to give this id a value. The alternative is that it has a value equal to the last function it generated, but I'm not sure that is useful, nor is it particularly intuitive.